src/edu/washington/escience/myria/parallel/Server.java

```diff
@@ -1057,7 +1066,51 @@ public DatasetStatus deleteDataset(final RelationKey relationKey) throws DbExcep
     }
 
     return getDatasetStatus(relationKey);
+  }
+
+  /**
+   * @param relationKey the relationKey of the dataset to persist
+   * @return the queryID
+   * @throws DbException if there is an error
+   * @throws InterruptedException interrupted
+   */
+  public long persistDataset(final RelationKey relationKey) throws DbException, InterruptedException {
+    long queryID;
+
+    /* Mark the relation as is_persistent */
+    try {
+      catalog.markRelationPersistent(relationKey);
+    } catch (CatalogException e) {
+      throw new DbException(e);
+    }
 
+    /* Create the query plan for persist */
+    try {
+      ImmutableMap<Integer, SubQueryPlan> workerPlans = null;
+      for (Integer workerId : getWorkersForRelation(relationKey, null)) {
+        String partitionName =
+            String.format(persistURI + "/partition-%s/%s/%s/%s", workerId, relationKey.getUserName(), relationKey
```

Comment:
Not sure what I suggested before, but it might be better to have a "reserved" prefix as a separate path component to denote this namespace, maybe something like `myria-system`? I don't have any real preferences here--maybe we could discuss for a couple minutes in the next Myria meeting and get consensus.